Fix rpmbuild-ctest to use also upstream Content

mildas opened the pull request:
Use `util.get_content` to get SSG content. If content is provided by env variable, use it, not the packaged one. Otherwise, use packaged content. In both cases, built it and test it using `cmake`.

comps commented on static-checks/rpmbuild-ctest/test.py:
```diff
@@ -1,52 +1,43 @@
+    # no default product, no derivatives, build only one specific rhel
+    cmd = [
+        'cmake', '../',
+        f'-DSSG_PRODUCT_RHEL{versions.rhel.major}:BOOLEAN=TRUE',
+        '-DSSG_PRODUCT_DEFAULT:BOOLEAN=FALSE',
+        '-DSSG_SCIENTIFIC_LINUX_DERIVATIVES_ENABLED:BOOL=OFF',
+        '-DSSG_CENTOS_DERIVATIVES_ENABLED:BOOL=OFF',
```
Does this really work with CentOS Stream (in CaC/content Github CI) ?  
(`SSG_PRODUCT_RHEL9.5` + `CENTOS_DERIVATIVES_ENABLED` OFF)

comps commented on static-checks/rpmbuild-ctest/test.py:
```diff
@@ -1,52 +1,43 @@
+        '-DSSG_ANSIBLE_PLAYBOOKS_PER_RULE_ENABLED:BOOL=ON',
+    ]
+    util.subprocess_run(cmd, cwd=build_dir)
+    cmd = ['make', '-j4']
+    util.subprocess_run(cmd, cwd=build_dir)
+    # ctest
```
Let's use some vertical space between these commands + let's not define `cmd` for one short command which can be inlined:
```suggestion

    util.subprocess_run(['make', '-j4'], cwd=build_dir)
    
    # ctest
```

comps commented on static-checks/rpmbuild-ctest/test.py:
```diff
@@ -1,52 +1,43 @@
+        for line in f:
+            result_match = result_regex.search(line)
+            if result_match:
+                test_name = result_match.group(1)
+                if result_match.group(2) == 'Passed':
```
```suggestion
                test_name, test_result = result_match.groups()
                if test_result == 'Passed':
```
?

comps commented on static-checks/rpmbuild-ctest/test.py:
```diff
@@ -1,52 +1,43 @@
+        for line in f:
+            result_match = result_regex.search(line)
+            if result_match:
+                test_name = result_match.group(1)
+                if result_match.group(2) == 'Passed':
+                    result = 'pass'
+                else:
+                    result = 'fail'
```
```suggestion
                test_name, test_result = result_match.groups()
                result = 'pass' if test_result == 'Passed' else 'fail'
```
?

comps commented on static-checks/rpmbuild-ctest/test.py:
```diff
@@ -1,52 +1,55 @@
+with util.get_content(build=False) as content_dir:
+    build_dir = content_dir / 'build'
+
+    util.subprocess_run(['cmake', '../', *cmake_options], cwd=build_dir)
+    util.subprocess_run(['make', '-j4'], cwd=build_dir)
```
These should also probably have `check=True`, which is not default.

comps commented on static-checks/rpmbuild-ctest/test.py:
```diff
@@ -1,52 +1,55 @@
+        'cmake', '--build', build_dir, '--target', 'test',
+        '--', 'ARGS=--output-on-failure --output-log ctest_results'
+    ]
+    util.subprocess_run(cmd, cwd=build_dir)
```
Same here, `check=True`.

mildas replied:
Not sure, I'll have to check it

mildas replied:
Return code 2 when any of unit tests fails, thus leaving this one without `check=True`

mildas commented:
@comps besides of added `check=True`, added also more `recommend` packages to enable more ssg unit tests

mildas commented:
Added new commit to fix search for failed ctests.

Previously, failed and passed results had same format:
```
40/52 Test #40: cce_avail_check-sle15 ............................................   Passed    0.04 sec
41/52 Test #41: utils-build_control_from_reference_sanity ........................   Failed   11.75 sec
```
but now there is:
```
40/52 Test #40: cce_avail_check-sle15 ............................................   Passed    0.04 sec
41/52 Test #41: utils-build_control_from_reference_sanity ........................***Failed   11.75 sec
```
and the regex didn't account for that as it expected spaces after `....`.

Thus, I've changed that to match only the part before result:
`40/52 Test #40: cce_avail_check-sle15 ............................................`
and not to match result. Just find the line and then search for `Passed` in it.

mildas commented:
@comps There are new packages added in `main.fmf` and some python3 modules are installed via `pip3` in test code. I want to enable as much as possible and as those are not available via dnf, I've chosen `pip3`-way. And intentionally without `check=True` in case any of the modules would be unavailable.

Tested on rhel8 and rhel9, passes. Thus, ready for merge.

comps commented on static-checks/rpmbuild-ctest/main.fmf:
```diff
@@ -6,10 +6,16 @@ environment+:
 duration: 1h
 require+:
```
You are also missing `python3-pip`.

comps replied:
Actually, in RHEL, `python3` depends on `python3-pip`, so we're good, .... but non-default python versions don't, ie. `python3.12` installs by default without `python3.12-pip`, so maybe I would still add it to keep track of that dependency explicitly.

Your choice.
